libcaf_test/caf/test/runnable.hpp

```diff
@@ -103,6 +175,26 @@ class CAF_TEST_EXPORT runnable {
   }
 #endif
 
+  template <class Expr>
+  void should_fail(Expr&& expr, const caf::detail::source_location& location
+                                = caf::detail::source_location::current()) {
+    auto* rep = reporter::instance;
+    auto before = rep->test_stats();
+    expr();
+    auto after = rep->test_stats();
+    check_eq(before.passed, after.passed, location);
+    if (check_eq(before.failed + 1, after.failed, location))
+      rep->test_stats({before.passed + 1, before.failed});
+  }
+
+  template <class T0, class T1>
+  static void assert_save_comparison() {
```

Comment:
It seems like you've missed this part from my earlier comment:

> I think this should be factored out into a (**private**) function.

This is a pure implementation detail and should not be part of the public API.

Reply:
Yes, sorry I seem to have missed it. I have updated the PR accordingly.